### PR TITLE
move faas to registry

### PR DESCRIPTION
**Pull request by trisch-me**
## Changes

Move faas attributes to the registry

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* [ ] ~~[schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.~~

### CODE REVIEW

File: model/registry/faas.yaml
Comment (joaopgrassi) [inline model/registry/faas.yaml:7]: Hum these was used as a resource attribute before. With the problem we faced in the tooling when generating code from it, I think we need to mark this as type: resource still. fyi @AlexanderWert

File: docs/faas/faas-spans.md
Comment (joaopgrassi) [inline docs/faas/faas-spans.md:166]: The requirement level for these is not the same anymore.

**Review by joaopgrassi — CHANGES_REQUESTED** [review]

File: model/registry/faas.yaml
Comment (AlexanderWert) [reply model/registry/faas.yaml:7]: I think we should block PRs that move a namespace to the registry, that contains both semantic attributes and resource attributes until we found a proper solution for this in the tooling.

**Comment by AlexanderWert** [comment]
Blocked by: https://github.com/open-telemetry/build-tools/issues/242

**Comment by joaopgrassi** [comment]
>Blocked by: https://github.com/open-telemetry/build-tools/issues/242

@AlexanderWert Now thinking, is this really be blocked? The attributes that were defined as resource before are not used on spans or metrics, so it should be fine to define them in the registry as resource for now, no? That would unblock this PR.

**Comment by AlexanderWert** [comment]
> > Blocked by: [open-telemetry/build-tools#242](https://github.com/open-telemetry/build-tools/issues/242)
> 
> @AlexanderWert Now thinking, is this really blocked? The attributes that were defined as resource before are not used on spans or metrics, so it should be fine to define them in the registry as resource for now, no? That would unblock this PR.

The problem is not only on single attributes but even when a single `namespace` contains a mix of resource and semantic attributes, which is the case here (for example: `faas.trigger`, `faas.coldstart`, `faas.invoked_*`, etc.)

**Comment by trisch-me** [comment]
I would vote for a proper workaround assuming we could easily merge it later. 
Also as we have discussed before, not all attributes can be easily divided into groups so I would like to have in registry `just` attributes.

Also as another option - we could merge all open PRs with `resource` attributes if needed and have a separate issue where we track it to change it later for all. We already do have temporary fix for some other attributes. This will reduce the amount of blocked PRs for now and should be fixed with just 1 PR later.

What do you think @AlexanderWert @joaopgrassi ?

**Comment by joaopgrassi** [comment]
>A workaround could be to define two attribute groups in registry, with the same prefix, where one would be for the resources and the other for semantic attributes.

>Also as another option - we could merge all open PRs with resource attributes if needed and have a separate issue where we track it to change it later for all. We already do have temporary fix for some other attributes. This will reduce the amount of blocked PRs for now and should be fixed with just 1 PR later.

Yeah that's what I meant, for those that don't appear in both spans/metrics and resources, we can just port them to the registry as they are today (with res ones having the same prefix but type resource) and the ones who are really appearing we block. At least then we get rid of all the blocked PRs, and with issues we can later track them down once the fix is available. 

Either way is fine by me

**Comment by trisch-me** [comment]
@AlexanderWert do you think I could change it to `resource` field attributes and then later we would change it to proper value as soon as fix it there?

File: docs/attributes-registry/faas.md
Comment (trisch-me) [inline docs/attributes-registry/faas.md:31]: Not a part of current refactoring but for maybe future PR
I see here a duplication of data for `invoked_name`, `invoked_provider` and `invoked_region`. Was it needed because of strict separation between resource and not resource attributes? Should we remove it now and replace with original fields as we are going into removing this restriction in build tools?

**Comment by trisch-me** [comment]
Actually I see we have here again mixed fields, from attributes and from resource, so it will break no matter what we will choose as the type. So we need to have build tools fixed in order to merge this one

**Comment by trisch-me** [comment]
@joaopgrassi could you please check again?

File: .chloggen/faas_registry.yaml
Comment (joaopgrassi) [inline .chloggen/faas_registry.yaml:7]: I think we can skip a changelog for this, as nothing really changed. Please remove it

**Review by joaopgrassi — APPROVED** [review]

**Review by AlexanderWert — APPROVED** [review]

**Review by jsuereth — APPROVED** [review]

File: docs/attributes-registry/faas.md
Comment (joaopgrassi) [reply docs/attributes-registry/faas.md:31]: @trisch-me could you please open an issue to track your concerns here? Then we can merge this and not forget to follow up later.

**Review by arminru — APPROVED** [review]

**Review by arminru — APPROVED** [review]

File: docs/attributes-registry/faas.md
Comment (lmolkova) [reply docs/attributes-registry/faas.md:31]: Created #799

**Review by lmolkova — APPROVED** [review]

**Pull request merged**
